crates/editor/src/editor.rs

```diff
@@ -12230,6 +12285,9 @@ impl Editor {
                 self.scrollbar_marker_state.dirty = true;
                 self.active_indent_guides_state.dirty = true;
                 self.refresh_active_diagnostics(cx);
+                if self.show_diagnostics_inline_enabled {
+                    self.start_show_diagnostics_inline(cx);
+                }
```

Comment:
Undo behaves quite oddly, apparently aggregating the lines into the same "group":

https://github.com/user-attachments/assets/f02bf6f5-d439-4ade-aaf6-e01af8e8eda8

Reply:
Huh. How many times did you save in there (or did clippy update). I don't have a good theory for why the diagnostics would behave that way. Though I'm fairly certain it hasn't got anything to do with how I'm rendering them, just that its yet another example on why we should probably have a pre-processing step separate from rendering.

Reply:
> How many times did you save in there (or did clippy update)

IIRC, only after duplicating the lines, I definitely did not save between undo/redo frequent calls shown in the late part of the video.

I presume this is related to `DisplayPoint` management and do not insist on fixing it right now, but needed to mention this.